### ISSUE TITLE
Issues inside array are not included in error if array has an issue

### ISSUE THREAD

**Issue by mymattcarroll** [colinhacks/zod#569]
Given the following schema and the data to validate:

```ts
const schema = z.object({
  people: z
    .object({
      firstName: z.string(),
      lastName: z.string(),
    })
    .array()
    .min(2),
})

const result = schema.safeParse({
  people: [
    {
      firstName: 'Me',
    },
  ],
})
```

I was hoping `result.error.issues` would be:

```json 
[
  {
    "code": "too_small",
    "minimum": 2,
    "type": "array",
    "inclusive": true,
    "message": "Should have at least 2 items",
    "path": [
      "people"
    ]
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "people",
      0,
      "lastName"
    ],
    "message": "Required"
  }
]
```

The actual `result.error.issues` is:

```json 
[
  {
    "code": "too_small",
    "minimum": 2,
    "type": "array",
    "inclusive": true,
    "message": "Should have at least 2 items",
    "path": [
      "people"
    ]
  }
]
```

**Comment by colinhacks** [comment]
I agree, this should behave the way you expect. Fixed in zod@3.7.0.

**Comment by mymattcarroll** [comment]
Wow!

Thank you for the quick response and even quicker fix.

Great library!
